core/src/main/java/org/jboss/pnc/causeway/source/SourceRenamer.java

```diff
@@ -0,0 +1,107 @@
+package org.jboss.pnc.causeway.source;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.DigestOutputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.commons.compress.compressors.CompressorException;
+import org.apache.commons.compress.compressors.CompressorStreamFactory;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream;
+import org.apache.commons.io.IOUtils;
+import org.jboss.pnc.causeway.CausewayException;
+
+@ApplicationScoped
+public class SourceRenamer {
+
+    public static final String ARCHIVE_SUFFIX = "-project-sources.tar.gz";
+    private CompressorStreamFactory compressor = new CompressorStreamFactory();
+
+    /**
+     * Repackage the sources archive while renaming the root directory to be named
+     * {@code <artifactId>-<version>-project-sources.tar.gz}.
+     */
+    public RenamedSources repackMaven(InputStream input, String groupId, String artifactId, String version)
+            throws CausewayException {
+        String gid = groupId.replace(".", "/");
+        String name = artifactId + "-" + version;
+        Path path = Paths.get(gid).resolve(artifactId).resolve(version);
+        return repack(input, name, path);
+    }
+
+    /**
+     * Repackage the sources archive while renaming the root directory to be named
```

Comment:
Have you meant `remaining` instead of `renaming` or did you forgot to delete `to be named`? I am not sure I understand.